src/test/java/no/unit/nva/cristin/projects/FetchOneCristinProjectTest.java

```diff
@@ -93,22 +106,28 @@ void handlerReturnsBadGatewayExceptionWhenFetchFromBackendFails() throws Excepti
 
         assertEquals(HttpURLConnection.HTTP_BAD_GATEWAY, response.getStatusCode());
         assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
 
-        doThrow(new FileNotFoundException()).when(cristinApiClientStub).getProject(any(), any());
+    @Test
+    void handlerReturnsServerErrorExceptionWhenBackendThrowsUriSyntaxException() throws Exception {
+        cristinApiClientStub = spy(cristinApiClientStub);
+
+        doThrow(URISyntaxException.class).when(cristinApiClientStub).getProject(any(), any());
         handler = new FetchOneCristinProject(cristinApiClientStub, environment);
         GatewayResponse<NvaProject> nextResponse = sendQueryWithId(DEFAULT_ID);
 
-        assertEquals(HttpURLConnection.HTTP_BAD_GATEWAY, response.getStatusCode());
-        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+        assertEquals(HttpURLConnection.HTTP_INTERNAL_ERROR, nextResponse.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, nextResponse.getHeaders().get(HttpHeaders.CONTENT_TYPE));
     }
 
     @Test
     void handlerReturnsNvaProjectWithoutParticipantsAndCoordinatingInstitutionIfTheyAreMissingFromBackend()
         throws Exception {
 
         cristinApiClientStub = spy(cristinApiClientStub);
```

Comment:
Although it is not part of this PR,  I think what we should be mocking is the httpclient. But we should create a task for this for doing it later